Packs/AzureRiskyUsers/Integrations/AzureRiskyUsers/AzureRiskyUsers.py

```diff
@@ -0,0 +1,395 @@
+
+        Args:
+            id (str): ID of risk detection to retrieve.
+
+        Return:
+            Response (dict): API response from AzureRiskyUsers.
+        """
+        return self.ms_client.http_request(method='GET',
+                                           url_suffix=f'/identityProtection/riskDetections/{id}')
+
+
+def build_query_filter(risk_state: Optional[str], risk_level: Optional[str]) -> Optional[str]:
+    """
+    Build query filter for API call, in order to get filtered results.
+
+    Args:
+        risk_state (str): Wanted risk state for filter.
+        risk_level (str): Wanted risk level for filter.
+
+    Returns:
+        str: Query filter string for API call.
+    """
+    if risk_state and risk_level:
+        return f"riskState eq '{risk_state}' and riskLevel eq '{risk_level}'"
+    elif risk_state:
+        return f"riskState eq '{risk_state}'"
+    elif risk_level:
+        return f"riskLevel eq '{risk_level}'"
+    else:
+        return None
+
+
+def create_event_or_incident_output(item: Dict,
+                                    table_headers: List[str]) -> Dict[str, Optional[Any]]:
+    """
+    Create the complete output dictionary for events or incidents.
+
+    Args:
+        item (dict): A source dictionary from the API response.
+        table_headers (list(str)): The table headers to be used when creating initial data.
+
+    Returns:
+        object_data (dict(str)): The output dictionary.
+    """
+    return remove_empty_elements({field: item.get(field) for field in table_headers})
+
+
+def risky_users_list_command(client: Client, args: Dict[str, str]) -> CommandResults:
+    """
+    List all risky users.
+    Args:
+        client (Client): Azure Risky Users API client.
+        args (dict): Arguments for API call.
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    limit = int(args['limit'])
+    page = int(args['page'])
+    risk_state = args.get('risk_state')
+    risk_level = args.get('risk_level')
+    skip_token = None
+    readable_message = f'Risky Users List\nCurrent page size: {limit}\nShowing page {page} out others that may exist'
+
+    if page > 1:
+        offset = limit * (page - 1)
+        raw_response = client.risky_users_list(risk_state,
+                                               risk_level,
+                                               offset)
+
+        next_link = raw_response.get('@odata.nextLink')
+        if not next_link:
+            return CommandResults(outputs_prefix='AzureRiskyUsers.RiskyUser',
+                                  outputs_key_field='id',
+                                  outputs=[],
+                                  readable_output=readable_message,
+                                  raw_response=[])
+        else:
+            parsed_url = urlparse(next_link)
+            skip_token = parse_qs(parsed_url.query)['$skiptoken'][0]
```

Comment:
this logic isn't very readable ATM.
Let's rewrite it to be more clear and easy to understand.
Moreover, can you explain what you're going for here?

Reply:
This logic is due to missing functionality for the Risky Users API endpoints, which is the option to query with page ($skip for the API) argument, so I had to use skip token instead, which made it more complicated.
Basically if the 'page' argument is specified it will call the API twice, first time it will get the unwanted results with the next link, and the second call will only get the relevant results with the next link retrieved from the first call.